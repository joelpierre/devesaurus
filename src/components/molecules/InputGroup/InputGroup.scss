.input-group {
  display: flex;
  width: 100%;
  align-items: stretch;

  &__prepend,
  &__append {
    display: inline-flex;
    align-items: center;
    height: 100%;
    width: auto;
  }

  &__inner {
  }

  &__input {
  }
}
