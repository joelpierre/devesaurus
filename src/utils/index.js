export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
