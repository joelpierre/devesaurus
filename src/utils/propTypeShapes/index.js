import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
