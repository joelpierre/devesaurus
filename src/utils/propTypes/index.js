import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
