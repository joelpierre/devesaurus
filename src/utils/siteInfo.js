import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
