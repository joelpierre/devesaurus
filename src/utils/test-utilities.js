import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
