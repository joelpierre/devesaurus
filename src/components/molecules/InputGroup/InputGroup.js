import React from 'react';
import PropTypes from 'prop-types';
import InputField from '../../atoms/InputField/InputField';
import Button from '../Buttons/Button';
import SvgIcon from '../../atoms/SvgIcon/SvgIcon';
import styles from './InputGroup.module.scss';
import { themePropType } from '../../../utils/propTypes';

const InputGroup = ({
  append,
  prepend,
  name,
  placeholder,
  disabled,
  readonly,
  min,
  max,
  type,
}) => {
  let inputField;
  let preButtonEl;
  let appButtonEl;
  let preIconEl;
  let appIconEl;
  let preText;
  let appText;

  if (prepend) {
    /**
     * If there is a button to prepend to the input group.
     */
    if (prepend.button) {
      const preButton = prepend.button;
      preButtonEl = (
        <Button
          link={preButton.link}
          text={preButton.text}
          behavior={preButton.behavior}
          size={preButton.size}
          theme={preButton.theme}
        />
      );
    }

    /**
     * If there is an icon to prepend to the input group
     */
    if (prepend.icon) {
      const preIcon = prepend.icon;
      preIconEl = <SvgIcon name={preIcon} />;
    }

    /**
     * If there is text to prepend
     */
    if (prepend.text) {
      preText = prepend.text;
    }
  }

  if (append) {
    /**
     * If there is a button to append to the input group
     */
    if (append.button) {
      const appButton = append.button;
      appButtonEl = (
        <Button
          link={appButton.link}
          text={appButton.text}
          behavior={appButton.behavior}
          size={appButton.size}
          theme={appButton.theme}
        />
      );
    }

    /**
     * If there is an Icon to append to the input group
     */
    if (append.icon) {
      const appIcon = append.icon;
      appIconEl = <SvgIcon name={appIcon} />;
    }

    /**
     * If there is text to append
     */
    if (append.text) {
      appText = append.text;
    }
  }

  /**
   * Switch statement to render the correct Input
   */
  switch (type) {
    case 'text':
    case 'email':
    case 'password':
    case 'date':
      inputField = (
        <InputField
          type={type}
          name={name}
          placeholder={placeholder}
          readOnly={readonly}
          disabled={disabled}
        />
      );
      break;
    case 'number':
      inputField = (
        <InputField
          type={type}
          name={name}
          minLength={min}
          maxLength={max}
          placeholder={placeholder}
          readOnly={readonly}
          disabled={disabled}
        />
      );
      break;
  }

  return (
    <div data-test="component-input-group" className={styles['input-group']}>
      {prepend && (
        <div className={styles['input-group__prepend']}>
          {preButtonEl && (
            <div
              data-test="input-group-button"
              className={styles['input-group__button-wrapper']}
            >
              {preButtonEl}
            </div>
          )}

          {preIconEl && (
            <div className={styles['input-group__inner']}>
              <div
                data-test="input-group-icon"
                className={styles['input-group__icon-wrapper']}
              >
                {preIconEl}
              </div>
            </div>
          )}

          {preText && (
            <div className={styles['input-group__inner']}>
              <div
                data-test="input-group-text"
                className={styles['input-group__text']}
              >
                {preText}
              </div>
            </div>
          )}
        </div>
      )}

      <div
        data-test="input-group-input"
        className={styles['input-group__input']}
      >
        {inputField}
      </div>

      {append && (
        <div className={styles['input-group__append']}>
          {appButtonEl && (
            <div
              data-test="input-group-button"
              className={styles['input-group__button-wrapper']}
            >
              {appButtonEl}
            </div>
          )}

          {appIconEl && (
            <div className={styles['input-group__inner']}>
              <div
                data-test="input-group-icon"
                className={styles['input-group__icon-wrapper']}
              >
                {appIconEl}
              </div>
            </div>
          )}

          {appText && (
            <div className={styles['input-group__inner']}>
              <div
                data-test="input-group-text"
                className={styles['input-group__text']}
              >
                {appText}
              </div>
            </div>
          )}
        </div>
      )}
    </div>
  );
};

InputGroup.defaultProps = {
  append: null,
  prepend: null,
  placeholder: null,
  disabled: false,
  readonly: false,
  min: null,
  max: null,
  type: 'text',
};

InputGroup.propTypes = {
  append: PropTypes.shape({
    button: PropTypes.shape({
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired,
      ...themePropType,
      behavior: PropTypes.oneOf(['action']),
      size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl', 'full']),
    }),
    icon: PropTypes.string,
    text: PropTypes.string,
  }),
  prepend: PropTypes.shape({
    button: PropTypes.shape({
      text: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired,
      ...themePropType,
      behavior: PropTypes.oneOf(['action']),
      size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl', 'full']),
    }),
    icon: PropTypes.string,
    text: PropTypes.string,
  }),
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['text', 'password', 'date', 'email', 'number']),
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  readonly: PropTypes.bool,
  min: PropTypes.string,
  max: PropTypes.number,
};

export default InputGroup;
