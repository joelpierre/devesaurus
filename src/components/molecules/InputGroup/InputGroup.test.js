import React from 'react';
import { shallow } from 'enzyme';

import InputGroup from './InputGroup';
import { findByTestAttr } from '../../../utils/test-utilities';

const defaultProps = {
  name: 'test-name',
  type: 'text',
};

/**
 * Factory function to create a ShallowWrapper for the InputGroup component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<InputGroup {...setupProps} />);
};

describe('<InputGroup/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Renders the InputGroup Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-input-group');
    expect(component.length).toBe(1);
  });
});
