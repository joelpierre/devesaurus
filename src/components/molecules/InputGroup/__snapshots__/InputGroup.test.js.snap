// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputGroup/> should match snapshot 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
</div>
`;

exports[`<InputGroup/> should match snapshot with append button 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
  <div
    className="input-group__append"
  >
    <div
      className="input-group__button-wrapper"
      data-test="input-group-button"
    >
      <Button
        action={null}
        behavior="action"
        caps={false}
        classes={null}
        link="#"
        size="md"
        text="test"
        theme="tint-omega"
      />
    </div>
  </div>
</div>
`;

exports[`<InputGroup/> should match snapshot with append icon 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
  <div
    className="input-group__append"
  >
    <div
      className="input-group__inner"
    >
      <div
        className="input-group__icon-wrapper"
        data-test="input-group-icon"
      >
        <SvgIcon
          center={false}
          classes={null}
          description="An SVG Icon"
          fill="#000000"
          name="complete"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<InputGroup/> should match snapshot with append text 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
  <div
    className="input-group__append"
  >
    <div
      className="input-group__inner"
    >
      <div
        className="input-group__text"
        data-test="input-group-text"
      >
        test
      </div>
    </div>
  </div>
</div>
`;

exports[`<InputGroup/> should match snapshot with prepend button 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__prepend"
  >
    <div
      className="input-group__button-wrapper"
      data-test="input-group-button"
    >
      <Button
        action={null}
        behavior="action"
        caps={false}
        classes={null}
        link="#"
        size="md"
        text="test"
        theme="tint-omega"
      />
    </div>
  </div>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
</div>
`;

exports[`<InputGroup/> should match snapshot with prepend icon 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__prepend"
  >
    <div
      className="input-group__inner"
    >
      <div
        className="input-group__icon-wrapper"
        data-test="input-group-icon"
      >
        <SvgIcon
          center={false}
          classes={null}
          description="An SVG Icon"
          fill="#000000"
          name="complete"
        />
      </div>
    </div>
  </div>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
</div>
`;

exports[`<InputGroup/> should match snapshot with prepend text 1`] = `
<div
  className="input-group"
  data-test="component-input-group"
>
  <div
    className="input-group__prepend"
  >
    <div
      className="input-group__inner"
    >
      <div
        className="input-group__text"
        data-test="input-group-text"
      >
        test
      </div>
    </div>
  </div>
  <div
    className="input-group__input"
    data-test="input-group-input"
  >
    <InputField
      classes={null}
      disabled={false}
      label={null}
      lg={false}
      max={null}
      md={false}
      min={null}
      name="test-name"
      placeholder={null}
      readOnly={false}
      sm={false}
      type="text"
    />
  </div>
</div>
`;
